Sources/C++Workshop/Cards_helper.h

```diff
@@ -14,20 +14,33 @@ template <typename Deck> class Card;
 template <typename Impl>
 class Deck {
 protected:
-    using Cards = std::vector<std::reference_wrapper<Card<Impl>>>;
+    using Card = Card<Impl>;
```

Comment:
I personally find it confusing to have an alias `Card` match an existing type/class/symbol/something.  Would you agree that unique names could be less confusing?

Reply:
I'm generally of the opinion that a member type with the same name as some more general type is a nice way to make a "specialized version" of that. `Deck<Impl>::Card` is `Card<Impl>` specialized to that `Deck...`. 🤷